### DISCUSSION TITLE
What is the actual problem with long-running tasks via hs.task?

### DISCUSSION THREAD

**Discussion by Madd0g** [Hammerspoon/hammerspoon#3602] [Q&A]
I've recently been wrapping a few shell commands in `hs.task`, I've been playing with whisper.cpp and wrapped it in hs.task.

But the thing keeps crashing after a while.

I see this note in the docs:

> This is not intended to be used for processes which never exit. While it is possible to run such things with hs.task, it is not possible to read their output while they run and if they produce significant output, eventually the internal OS buffers will fill up and the task will be suspended.

I'm calling the whisper.cpp executable via /usr/local/bin/stdbuf (which chatgpt recommended to make sure I get frequent streaming messages).

The note says it's impossible to get their output while they run? but it works fine for me, I use a streaming callback and get the output live.

What does the note mean regarding buffers? Is it hammerspoon that's gathering all the output to pass it to the final callback? Or something else?

Tell me your tricks, are there workarounds?

Thanks!

**Comment by muescha** [comment]
when I hear about crashing after a while: maybe you need to hold a reference to this task as a global variable to avoid garbage collection. check for this at first try.
